sale_coupon_multi_currency/tests/test_sale_coupon_multi_currency.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from .common import TestSaleCouponMultiCurrencyCommon
+
+
+class TestSaleCouponMultiCurrency(TestSaleCouponMultiCurrencyCommon):
+    """Test custom currency on program."""
+
+    def test_01_program_multi_currency(self):
+        """Set custom currency.
+
+        Case 1: set custom currency other than company currency.
+        Case 2: unset custom currency to use company currency.
+        """
+        # Case 1.
+        self.program_coupon_percentage.currency_id = self.currency_other.id
+        self.assertEqual(
+            self.program_coupon_percentage.currency_id, self.currency_other
+        )
+        self.assertEqual(
+            self.program_coupon_percentage.currency_custom_id, self.currency_other
+        )
+        self.assertEqual(self.company_main.currency_id, self.currency_company)
+        # Case 2.
+        self.program_coupon_percentage.currency_custom_id = False
```

Comment:
`currency_custom_id` is normally not modified directly, but using inverse function on `currency_id`:
```suggestion
        self.program_coupon_percentage.currency_id = False
```

Reply:
Well this is test. It won't work, because then it unsets company currency. This is a test case where we look what would be computed if custom currency is unset. From interface, you can't unset that currency, so this is a bit different than mimicking `currency_id` inverse.